src/components/uielements/filterMenu/FilterMenu.tsx

```diff
@@ -3,30 +3,36 @@ import React, { useCallback, useMemo, useState } from 'react'
 import { SearchOutlined } from '@ant-design/icons'
 import { ClickParam } from 'antd/lib/menu'
 
-import { AssetPair } from '../../../types/asgardex'
 import { Input } from '../input'
 import { Menu, MenuItem } from './FilterMenu.style'
 
-type Props = {
+type Props<T> = {
   asset?: string
-  data: AssetPair[]
+  data: T[]
   placeholder?: string
   searchEnabled?: boolean
-  cellRenderer: (data: AssetPair) => { key: string; node: JSX.Element }
-  disableItemFilter?: (item: AssetPair) => boolean
-  filterFunction: (item: AssetPair, searchTerm: string) => boolean
+  cellRenderer: (data: T) => { key: string; node: JSX.Element }
+  disableItemFilter?: (item: T) => boolean
+  filterFunction: (item: T, searchTerm: string) => boolean
   onSelect?: (value: string) => void
 }
 
-const FilterMenu: React.FC<Props> = ({
-  onSelect = (_) => {},
-  searchEnabled = false,
-  data,
-  filterFunction,
-  cellRenderer,
-  disableItemFilter = (_) => false,
-  placeholder = 'Search Token ...'
-}): JSX.Element => {
+// Note: To have fully support of generic types in props,
+// we define FilterMenu as an "old-school" a function here.
+// Arrow functions seems to have some issues with generic types for properties in React.FC.
+// Based on "How to use generics in props in React in a functional component?"" https://stackoverflow.com/a/59373728/2032698
+function FilterMenu<T>(props: Props<T>): JSX.Element {
+  // const FilterMenu = <T extends object>(props: Props<T>): JSX.Element => {
```

Comment:
seems here is no any problems with using function expression.
you can see this error 
![image](https://user-images.githubusercontent.com/67793194/86955947-0bab8500-c161-11ea-804b-ede8b875e712.png)
only 'cause `T` should not extend object. It should extends or `unknown` or `any`. if we take a look at `FilterMenu.stories.tsx` we can see that data there is a string
![image](https://user-images.githubusercontent.com/67793194/86956620-0ac72300-c162-11ea-910d-895c88d5c8a2.png)

Reply:
How the function expression looks like in your case?

Reply:
@veado hm. just uncommented the line there were before

Reply:
oh no :) that was no the only one. also as i mentioned we had to do next:
![image](https://user-images.githubusercontent.com/67793194/87022710-916e1580-c1df-11ea-8e0f-8a523c794782.png)

Reply:
Thanks, fixed by 09b4083c342a514e497025755d9cddf6a435e076